content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-jenkins-identity/index.md

```diff
@@ -0,0 +1,426 @@
+        stage('oidc-test') {
+            steps {
+                withCredentials([string(variable: 'token', credentialsId: 'oidc-token')]) {
+                    sh '''
+                        wget -O chainctl "https://dl.enforce.dev/chainctl/latest/chainctl_linux_\$(uname -m)"
+                        chmod +x chainctl
+                        ./chainctl -v3 auth login --identity-token $token --identity <your jenkins identity>
+                    '''
+                }
+            }
+        }
+    }
+}
+```
+
+The important line is `withCredentials` option, which maps the generated OIDC token from the `oidc-token` credential parameter to `token` variable in the pipeline step.
+
+Now you can add the commands for testing the identity like `chainctl policy ls` in the following example:
+
+```
+sh '''
+    wget -O chainctl "https://dl.enforce.dev/chainctl/latest/chainctl_linux_\$(uname -m)"
+    chmod +x chainctl
+    ./chainctl -v3 auth login --identity-token $token --identity <your jenkins identity>
+    ./chainctl policy ls
+'''
+```
+
+Save the job, and then build it using the `Build with Parameters` option.
+
+Assuming everything works as expected, your pipeline will be able to assume the identity and run the `chainctl policy ls` command, returning the policy you created with the `sample.tf` file.
+
+```
+. . .
+
+chainctl        	100%[===================>]  54.34M  6.78MB/s	in 13s
+
+2023-05-17 13:19:45 (4.28 MB/s) - ‘chainctl’ saved [56983552/56983552]
+
+Successfully exchanged token.
+Valid! Id: 3f4ad8a9d5e63be71d631a359ba0a91dcade94ab/d3ed9c70b538a796
+                         	ID                         	| 	NAME  	| DESCRIPTION |   MODE
+------------------------------------------------------------+---------------+-------------+-----------
+  3f4ad8a9d5e63be71d631a359ba0a91dcade94ab/42b8649dc3a3ea66 | trust-any-cgr |         	| ENFORCED
+```
+
+If you'd like to experiment further with this identity and what the pipeline can do with it, there are a few parts of this setup that you can tweak. For instance, if you'd like to give this identity different permissions you can change the role data source to the role you would like to grant.
+
+```
+data "chainguard_roles" "editor" {
+  name = "editor"
+}
+```
+
+You can also edit the pipeline itself to change its behavior. For example, instead of inspecting the policies the identity has access to, you could have the workflow inspect the groups.
+
+```
+	. . .
+
+	- './chainctl iam groups ls'
+```
+
+Of course, the Jenkins pipeline will only be able to perform certain actions on certain resources, depending on what kind of access you grant it.
+
+
+## Removing Sample Resources
+
+To remove the resources Terraform created, you can run the `terraform destroy` command.
+
+```sh
+terraform destroy
+```
+
+This will destroy the sample policy, the role-binding, and the identity created in this guide. However, you'll need to destroy the `example-group` group yourself with `chainctl`.
+
+```sh
```

Comment:
so `terraform destroy` didn't destroy my policy or group, I had to delete them both manually. But I couldn't delete the group until the policy under was deleted first.

Reply:
That's an odd one, I'll give it another run through and check. If I can't get the policy to be deleted with terraform, at this point I'll just make a note to delete manually with chainctl.